packages/mdc-switch/foundation.js

```diff
@@ -0,0 +1,106 @@
+/**
+ * @license
+ * Copyright 2018 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import MDCFoundation from '@material/base/foundation';
+/* eslint-disable no-unused-vars */
+import {MDCSelectionControlState} from '@material/selection-control/index';
+import MDCSwitchAdapter from './adapter';
+/* eslint-enable no-unused-vars */
+import {cssClasses, strings} from './constants';
+
+/**
+ * @extends {MDCFoundation<!MDCSwitchAdapter>}
+ */
+class MDCSwitchFoundation extends MDCFoundation {
+  /** @return enum {string} */
+  static get strings() {
+    return strings;
+  }
+
+  /** @return enum {string} */
+  static get cssClasses() {
+    return cssClasses;
+  }
+
+  /** @return {!MDCSwitchAdapter} */
+  static get defaultAdapter() {
+    return /** @type {!MDCSwitchAdapter} */ ({
+      addClass: (/* className: string */) => {},
```

Comment:
I think it would yes, but I was just basing this on looking at most of the other foundation get defaultAdapter methods which do this and assumed this was the preferred style. Do you know if it's cool to move away from this now or if there's  a reason the other files are doing it?

Reply:
AFAIK, these are not closure annotations :)

We might be duplicating the defining the types in foundation & adapter file.